containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -42,52 +45,31 @@ spec:
                 echo "Error: Input cannot contain newline. Input: '$input_string'"
                 exit 1
             fi
+
+            if echo "$input_string" | grep -q -e '"' -e "'"; then
+              echo "Error: Input cannot contain quotes. Input: '$input_string'"
+              exit 1
+            fi
           }
 
           validate_proxy "${HTTP_PROXY:-}"
           validate_proxy "${HTTPS_PROXY:-}"
-          validate_proxy "${NO_PROXY:-NO_PROXY=localhost}"
+          validate_proxy "${NO_PROXY:-localhost}"
 
-          STARTUP_SCRIPT=$(cat <<END
-          set -e
-          set -u
-          mkdir -p /etc/systemd/system/containerd.service.d
-          echo "[Service]
-          Environment=${HTTP_PROXY}
-          Environment=${HTTPS_PROXY}
-          Environment=${NO_PROXY:-NO_PROXY=localhost}" > /etc/systemd/system/containerd.service.d/http-proxy.conf
+          cat > /etc/systemd/system/containerd.service.d/http-proxy.conf <<EOF
+          [Service]
+          Environment="HTTP_PROXY=${HTTP_PROXY}"
+          Environment="HTTPS_PROXY=${HTTPS_PROXY}"
+          Environment="NO_PROXY=${NO_PROXY:-localhost}"
+          EOF
+
+          EXEC="nsenter -t 1 -m -p --"
           echo "Reloading systemd management configuration"
-          systemctl daemon-reload
+          $EXEC sh -c "systemctl daemon-reload"
```

Comment:
nit: Do you need `sh -c` here? Can you just nsenter to execute `systemctl` directly?

Reply:
Right, we don't need sh -c here. Removed.